### ISSUE TITLE
Restic mount lists symlinks with size of zero

### ISSUE THREAD

**Issue by lenzj** [restic/restic#3667]
Output of `restic version`
--------------------------
restic 0.12.1 (v0.12.1-213-gdaea461f) compiled with go1.17.7 on linux/amd64

How did you run restic exactly?
-------------------------------
$ restic -r path/to/repo mount myfusedir

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Local hard disk

Expected behavior
-----------------
While using "ls -l" on files within the mounted fuse repository I would expect that symlink files would have sizes greater than zero.

Actual behavior
---------------
All symlink files are displayed with a size of zero when using "ls -l".  Most of the time it is not an issue accessing files, however it causes problems with the tarlz utility (https://lzip.nongnu.org/tarlz.html) that I recently started using.  tarlz is able to access and archive all files with the exception of symlink files in a restic fuse mount.  For symlink files it displays a read error.

If I copy the symlink files to my ext4 drive (cp -a path/to/restic/fuse path/to/ext4/) then tarlz has no problems accessing and archiving the symlink files.

Steps to reproduce the behavior
-------------------------------
Any restic repository containing symlinks will have this behavior in my experience.

Do you have any idea what may have caused this?
-----------------------------------------------
I'm assuming the fuse code within restic is incorrectly calculating (or ignoring) symlink file sizes.

Do you have an idea how to solve the issue?
-------------------------------------------
If I get a chance I will try and find the respective fuse code to see if I can find the source of the problem.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Restic is awesome, and yes it makes me happy.

**Comment by lenzj** [comment]
Per POSIX, the size of a symlink should be equal to the length in bytes of the pathname contained in the symlink.
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html
```
   off_t st_size           For regular files, the file size in bytes.
                           For symbolic links, the length in bytes of the
                           pathname contained in the symbolic link. 
```

It looks like this restic function below is the one of interest.  I will try adding an "a.Size" entry in that function and if all goes well submit a pull request.
https://github.com/restic/restic/blob/daea461f1575e860424bcbbae8cc7292ce3fb919/internal/fuse/link.go#L30

**Comment by lenzj** [comment]
@greatroar Thank you for your pull request.  I was working on this at the same time and had a similar change in the works.    I didn't think to include the block count update.  I tested your updates and tried accessing the fuse repo with the tarlz utility and it works great with and without the inclusion of the block count.  It's good insurance regardless so I would recommend keeping it in.

There was one minor issue remaining though.  The "latest" symlink in the higher level fuse folder structure also has a symlink size of zero.  I think a similar update is needed somewhere near the location below.

https://github.com/restic/restic/blob/daea461f1575e860424bcbbae8cc7292ce3fb919/internal/fuse/snapshots_dir.go#L308

**Comment by lenzj** [comment]
@greatroar I tried this patch below in addition to the changes you already made and that seems to solve the issue with sizes on the "latest" symlinks.  Can you include this in #3668 ?
```diff
diff --git a/internal/fuse/snapshots_dir.go b/internal/fuse/snapshots_dir.go
index 34484b59..4371f656 100644
--- a/internal/fuse/snapshots_dir.go
+++ b/internal/fuse/snapshots_dir.go
@@ -440,6 +440,8 @@ func (l *snapshotLink) Readlink(ctx context.Context, req *fuse.ReadlinkRequest)
 func (l *snapshotLink) Attr(ctx context.Context, a *fuse.Attr) error {
        a.Inode = l.inode
        a.Mode = os.ModeSymlink | 0777
+       a.Size = uint64(len(l.target))
+       a.Blocks = 1 + a.Size/blockSize
        a.Uid = l.root.uid
        a.Gid = l.root.gid
        a.Atime = l.snapshot.Time
```

**Comment by greatroar** [comment]
@lenzj Merged your patch into my #3668.

**Comment by fd0** [comment]
Thanks!
